File: src/main/java/com/damianryan/octopus/model/GasMeterPoint.java
package com.damianryan.octopus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class GasMeterPoint {

    String mprn;

    @JsonProperty("consumption_standard")
    int consumptionStandard;

    List<GasMeter> meters;

    List<Agreement> agreements;
}

File: src/main/kotlin/com/damianryan/octopus/model/GasMeterPoint.kt
package com.damianryan.octopus.model

import com.fasterxml.jackson.annotation.JsonProperty

data class GasMeterPoint(
    val mprn: String,
    @JsonProperty("consumption_standard") val consumptionStandard: Int = 0,
    val meters: List<GasMeter>,
    val agreements: List<Agreement>
)
